src/main/java/appeng/core/features/ItemFeatureHandler.java

```diff
@@ -105,15 +108,43 @@ else if( name.equals( "ItemPart" ) )
 
 			if( side == Side.CLIENT )
 			{
-				ModelBakery.registerItemVariants( item, registryName );
+				if( item instanceof AEBaseItem )
+				{
+					AEBaseItem baseItem = (AEBaseItem) item;
+
+					// Handle registration of item variants
+					ResourceLocation[] variants = baseItem.getItemVariants().toArray( new ResourceLocation[0] );
+					ModelBakery.registerItemVariants( item, variants );
+				}
+				else
+				{
+					ModelBakery.registerItemVariants( item, registryName );
+				}
```

Comment:
Implementing the method *is* optional. This is the default implementation of the method provided in AEBaseItem:
````
	public List<ResourceLocation> getItemVariants() {
		return Lists.newArrayList( getRegistryName() );
	}
````

And regarding the registry name: Why is that the case? I didn't find any evidence while tracing the call that the item registry name is actually a required variant name for items.